module/input/SensorFilter/src/MotionModel.hpp

```diff
@@ -158,22 +158,15 @@ namespace module::input {
             // Here is an explanation https://gamedev.stackexchange.com/a/157018
             // Here is another derivation https://fgiesen.wordpress.com/2012/08/24/quaternion-differentiation/
             // dq/dt = (1/2)*omega*Rwt
-            const Eigen::Quaternion<Scalar> dq_dt = Eigen::Quaternion<Scalar>(0.0,
-                                                                              newState.omegaTTt.x() * 0.5,
-                                                                              newState.omegaTTt.y() * 0.5,
-                                                                              newState.omegaTTt.z() * 0.5)
-                                                    * Rwt;
+            const Eigen::Matrix<Scalar, 4, 1> half_omegaTTt_coeffs(newState.omegaTTt.x() * 0.5,
+                                                                   newState.omegaTTt.y() * 0.5,
+                                                                   newState.omegaTTt.z() * 0.5,
+                                                                   0.0);
+            const Eigen::Quaternion<Scalar> dq_dt = Eigen::Quaternion<Scalar>(half_omegaTTt_coeffs) * Rwt;
```

Comment:
I would just wrap these straight into the constructor, e.g. `Quat(Vec(a,b,c,d));`